### ISSUE TITLE
Cleanup directory after generating plans

### ISSUE THREAD

**Issue by MFaisalZaki** [aiplan4eu/up-symk#3]
Can I ask to modify the code to remove the tmp directory after returning the generated plans? This will be helpful when running multiple configurations of SymK and asking for many plans. 

This is a simple fix:
rather than (with ... as ...)[https://github.com/aiplan4eu/up-symk/blob/3f9aa81a86fcb0dfb6628f7967192e1d0bdcb450/up_symk/symk.py#L204C47-L204C54] can be updated to be in this pattern
```
tmpdir = tempfile.TemporaryDirectory()
# Do the planning stuff

# Invoke cleanup after solving the planning task.
tmpdir.cleanup()
```

**Comment by speckdavid** [comment]
Hi all,

In general this sounds like a good change to me and the linked line is one place to change it. However, this line only affects the oversubscription planning mode of SymK. In fact, this code is a slightly modified version of the code in the superclass of `PDDLPlanner` in the unified planning (UP) package. I would call it a hack that serves its purpose.

Changing the code in the `up-symk` package would not affect the normal single plan and top-k search. Which mode do you currently use?

I think it would be best to change the code in the UP: [code-line](https://github.com/aiplan4eu/unified-planning/blob/d08701bf89887cd2250ccd73a441be86b18cc2f5/unified_planning/engines/pddl_planner.py#L175). Then all modes of SymK should have this behavior (without the osp search, which should be easy to update).

Can you open an issue in the [unified-planning](https://github.com/aiplan4eu/unified-planning/tree/master) repo?

**Comment by MFaisalZaki** [comment]
Hi @speckdavid, 

Thanks for highlighting this. I'm using the top-k version. I have opened an [issue here](https://github.com/aiplan4eu/unified-planning/issues/537)

**Comment by speckdavid** [comment]
Has the issue been resolved? I read that the files are only temporary created.

(By the way, we have released a new version of the up-symk package, 1.1.0, which fixes a problem with the python pipeline where action costs were sometimes ignored).

**Comment by MFaisalZaki** [comment]
@speckdavid Yup thanks a lot.
